json_object.c

```diff
@@ -1047,7 +1047,7 @@ static int json_object_double_to_json_string_format(struct json_object *jso, str
 		{
 			// Ensure it looks like a float, even if snprintf didn't,
 			//  unless a custom format is set to omit the decimal.
-			strcat(buf, ".0");
+			strncat(buf, ".0", 2);
```

Comment:
gcc 9 has a rather bogus warning on this line, but this is pointless change anyway and just serves to introduce potential discrepancies between the constant string and the length passed.